helm-v3/006-repositories.md

```diff
@@ -34,3 +35,135 @@ provide an implementation for.
+YAML.
+
+Third, there will be 2 types of files. The `index.json` file will hold the top
+level details about the repository. Alongside that there will be JSON files for
+each chart in the repository. These chart JSON files will contain the details
+about the revisions of the chart.
+
+The following two examples illustrate index and chart JSON files:
+
+```json
+{
+  "apiVersion": "v2",
+  "entries": {
+    "artifactory": {
+      "ref": "https://kubernetes-charts-incubator.storage.googleapis.com/artifactory.json",
+      "stable": {
+        "created": "2017-07-06T01:33:50.952Z",
+        "description": "Universal Repository Manager supporting all major packaging formats,\nbuild tools and CI servers.",
+        "digest": "249e27501dbfe1bd93d4039b04440f0ff19c707ba720540f391b5aefa3571455",
+        "home": "https://www.jfrog.com/artifactory/",
+        "icon": "https://raw.githubusercontent.com/JFrogDev/artifactory-dcos/master/images/jfrog_med.png",
+        "keywords": [
+          "artifactory",
+          "jfrog"
+        ],
+        "maintainers": [
+          {
+            "email": "[redacted]",
+            "name": "[redacted]"
+          }
+        ],
+        "name": "artifactory",
+        "sources": [
+          "https://bintray.com/jfrog/product/JFrog-Artifactory-Pro/view",
+          "https://github.com/JFrogDev"
+        ],
+        "urls": [
+          "https://kubernetes-charts-incubator.storage.googleapis.com/artifactory-5.2.0.tgz"
+        ],
+        "version": "5.2.0"
+      }
+    }
+  }
+}
+```
+
+The example above is of the `index.json` file. An entry here has two properties.
+The `ref` property is the location of the chart JSON file. This can be either
+a relative URL path relative to the `index.json` file or an absolute path to the
+file. The `latest` property contains the details about the latest release of the
+chart.
+
+The example below is the `artifactory.json` file referred to in the index. The
+information for each version of the is listed under versions. This is in a similar
+format to the repository v1 `entries` for a single chart. The different in format
+is that an object with versions as keys is used instead of an array as seen in
+the v1 specification.
+
+```json
+{
+  "apiVersion": "v2",
+  "versions": {
+    "5.2.0": {
+        "created": "2017-07-06T01:33:50.952Z",
+        "description": "Universal Repository Manager supporting all major packaging formats,\nbuild tools and CI servers.",
+        "digest": "249e27501dbfe1bd93d4039b04440f0ff19c707ba720540f391b5aefa3571455",
+        "home": "https://www.jfrog.com/artifactory/",
+        "icon": "https://raw.githubusercontent.com/JFrogDev/artifactory-dcos/master/images/jfrog_med.png",
+        "keywords": [
+            "artifactory",
+            "jfrog"
+        ],
+        "maintainers": [
+            {
+            "email": "[redacted",
+            "name": "[redacted"
+            }
+        ],
+        "name": "artifactory",
```

Comment:
Is there a need to namespace this further? Currently, the repository is acting as a namespace. Example - There are multiple variations of MongoDB or say RabbitMQ chart. They are all stored in different repositories. Does it make sense to create a namespace in case a single repository is used? Like for maven, there is groupId that also acts as a namespace. And then there is artifactId and version.

Reply:
Makes sense.